File: app/src/main/java/com/jimzrt/umsmounter/tasks/BaseTask.java
package com.jimzrt.umsmounter.tasks;


import android.app.Activity;

import java.lang.ref.WeakReference;

public abstract class BaseTask {

    //  Context ctx;
    String name;
    String description;
    String result;
    WeakReference<Activity> ctx;


    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }


    public abstract boolean execute();

    public String getResult() {
        return result;
    }

    public void setContext(WeakReference<Activity> context) {
        this.ctx = context;
    }
}

File: app/src/main/java/com/jimzrt/umsmounter/tasks/BaseTask.kt
package com.jimzrt.umsmounter.tasks

import android.app.Activity
import java.lang.ref.WeakReference

abstract class BaseTask {
    //  Context ctx;
    @JvmField
    var name: String? = null

    @JvmField
    var description: String? = null

    @JvmField
    var result: String? = null

    @JvmField
    var ctx: WeakReference<Activity>? = null

    abstract fun execute(): Boolean

    fun setContext(context: WeakReference<Activity>?) {
        ctx = context
    }
}